packages/app/src/cli/services/app-logs/logs-command/poll-app-logs.ts

```diff
@@ -0,0 +1,43 @@
+import {PollOptions, AppLogData} from '../types.js'
+import {fetchAppLogs} from '../utils.js'
+import {AbortError} from '@shopify/cli-kit/node/error'
+
+export const pollAppLogsForLogs = async ({
+  jwtToken,
+  cursor,
+  filters,
+}: PollOptions): Promise<{
+  cursor?: string
+  errors?: {
+    status: number
+    message: string
+  }[]
+  appLogs?: AppLogData[]
+}> => {
+  const response = await fetchAppLogs(jwtToken, cursor, filters)
+
+  const responseJson = await response.json()
+  if (!response.ok) {
+    const errorResponse = responseJson as {
+      errors: string[]
+    }
+    if (response.status === 401 || response.status === 429 || response.status >= 500) {
+      return {
+        errors: [{status: response.status, message: `${errorResponse.errors.join(', ')}`}],
+      }
+    } else {
+      throw new AbortError(`${errorResponse.errors.join(', ')} while fetching app logs`)
+    }
+  }
+
+  const data = responseJson as {
+    app_logs?: AppLogData[]
+    cursor?: string
+    errors?: string[]
```

Comment:
In this success case, are `app_logs` and `cursor` always present? Also do we need to type `errors`?